### PR TITLE
CoDICE L1a lo-nsw-species-counts

### LINKED ISSUE #252: Produce CoDICE L1A CDFs
Remaining work:

- [x] #767 
- [x] #768 
- [x] #769 
- [x] #770 
- [x] #771 
- [x] #772 
- [x] #773 
- [x] #774 
- [x] #775 
- [x] #776 
- [x] #785 
- [x] #809 
- [x] #992 
- [x] #1154 
- [x] #1155

### PR DESCRIPTION
**Pull request by bourque**
# Change Summary

## Overview

This PR adds support for `lo-nsw-species-counts` data and creates corresponding CDF. It also includes some minor updates on APID names.

## New Files

- `imap_processing/tests/codice/data/lo_fsw_view_6_ccsds.bin`
    - Test data for `lo-nsw-species-counts` APID 

## Updated Files

- `imap_processing/codice/cdf_attrs.py`
    - Added CDF attrs for `lo-nsw-species-counts` data
- `imap_processing/codice/codice_l0.py` 
    - Added mapping to `lo-nsw-species-counts` packet definition
- `imap_processing/codice/codice_l1a.py`
    - Added support for `lo-nsw-species-counts` APID, and some updates to make some functions/methods a bit more generalized 
- `imap_processing/codice/constants.py` 
    - Added CDF-friendly names for `lo-sw-species-counts`
- `imap_processing/codice/packet_definitions/P_COD_LO_NSW_SPECIES_COUNTS.xml`
    - Packet definition now uses variable length data   
- `imap_processing/codice/utils.py`
    - Added/updated some APIDs and their names
- `imap_processing/tests/codice/test_codice_l1a.py`   
    - Added test for `lo-nsw-species-counts` CDF

### CODE REVIEW

File: imap_processing/tests/codice/test_codice_l1a.py
Comment (greglucas) [inline imap_processing/tests/codice/test_codice_l1a.py:1]: Could you test the shapes / sizes expected from your test data to make sure we are getting the right dimensions of products produced from the test files?

Do you have the actual data values that you want to compare for the output too with some `np.testing.assert_array_equal(...)` from values you have gotten from CoDICE?

**Review by greglucas — APPROVED** [review]
I like that you kept this to a single data product, very manageable to review :) The code abstractions look good to me, but it would be nice to get another test associated with the uploaded test data and not just making sure the files are created if that is easy to do.

File: imap_processing/codice/cdf_attrs.py [hunk 1]
```diff
@@ -62,6 +62,13 @@
     instrument_base=codice_base,
 )
 
+l1a_lo_nsw_species_attrs = GlobalDataLevelAttrs(
+    data_type="L1A_SCI->Level-1A Science Data",
+    logical_source="imap_codice_l1a_lo-nsw-species-counts",
+    logical_source_desc="IMAP Mission CoDICE Instrument Level-1A Lo Non-sunward Species Data",  # noqa
```
Comment (tech3371) [inline imap_processing/codice/cdf_attrs.py]: If this line gives "line too long error", I believe you can do this
```suggestion
    logical_source_desc=(
                 "IMAP Mission CoDICE Instrument Level-1A "
                 "Lo Non-sunward Species Data"
                 ),
```

File: imap_processing/tests/codice/test_codice_l1a.py
Comment (bourque) [reply imap_processing/tests/codice/test_codice_l1a.py:1]: I've been thinking about ways to expand the test suite. This is a good idea.  I don't have validation data/files from CoDICE yet, so I don't yet have anything to compare the array values to.

I think for the time being though I can set up some testing framework for this, especially for comparing the shapes.  The values might change once I acquire some validation data, but that hopefully would just be a simple update to the 'truth' values.

I will tackle this in my next PR, so we can keep the PRs small!

**Review by tech3371 — APPROVED** [review]

**Comment by bourque** [comment]
Related to #252

**Pull request merged**